File: impl/cpp/Node.cpp
using namespace std;

#include "Item.h"
#include "Node.h"

template<typename T>
Item<T> Node<T>::getItems() {
    return items;
}

template<typename T>
void Node<T>::setItems(Item<T> item) {
    items = item;
}

template<typename T>
T Node<T>::getKey() {
    return key;
}

template<typename T>
void Node<T>::setKey(T k) {
    key = k;
}

template<typename T>
int Node<T>::getRank() {
    return rank;
}

template<typename T>
void Node<T>::setRank(int r) {
    rank = r;
}

template<typename T>
Node<T> Node<T>::getLeft() {
    return left;
}

template<typename T>
void Node<T>::setLeft(Node<T> l) {
    left = l;
}

template<typename T>
Node<T> Node<T>::getRight() {
    return right;
}

template<typename T>
void Node<T>::setRight(Node<T> r) {
    right = r;
}

template<typename T>
Node<T> Node<T>::getNext() {
    return next;
}

template<typename T>
void Node<T>::setNext(Node<T> n) {
    next = n;
}

File: impl/java/src/softheap/Node.java
package softheap;

public class Node<T extends Comparable<T>> {
    private Item<T> items;
    private T key;
    private int rank;
    private Node<T> left, right, next;

    public Item<T> getItems() {
        return items;
    }

    public void setItems(Item<T> item) {
        items = item;
    }

    public T getKey() {
        return key;
    }

    public void setKey(T key) {
        this.key = key;
    }

    public int getRank() {
        return rank;
    }

    public void setRank(int rank) {
        this.rank = rank;
    }

    public Node<T> getLeft() {
        return left;
    }

    public void setLeft(Node<T> left) {
        this.left = left;
    }

    public Node<T> getRight() {
        return right;
    }

    public void setRight(Node<T> right) {
        this.right = right;
    }

    public Node<T> getNext() {
        return next;
    }

    public void setNext(Node<T> next) {
        this.next = next;
    }
}
